client/src/partials/vouchers/simple.js

```diff
@@ -2,116 +2,73 @@ angular.module('bhima.controllers')
 
 SimpleJournalVoucherController.$inject = [
-  'AppCache', 'VoucherService', '$translate', 'AccountService',
-  'CurrencyService', 'SessionService'
+  'AppCache', 'VoucherService', 'AccountService', 'SessionService',
+  'NotifyService'
 ];
 
 /**
- * Simple Journal Vouchers
+ * @class SimpleJournalVouchers
  *
  * This module implements simply journal vouchers, crafted specially for cash
- * transactions, but useable in any generic transactions that only require two
+ * transactions, but usable in any generic transactions that only require two
  * Posting Journal lines.  It allows users to quickly create transactions by
  * specifying two accounts and an amount to transfer between them.
  *
- * @constructor
- *
  * @todo - Implement caching mechanism for incomplete forms (via AppCache)
  * @todo - Implement Voucher Templates to allow users to save pre-selected
  * forms (via AppCache and the breadcrumb component).
  */
-function SimpleJournalVoucherController(AppCache, Vouchers, $translate, Accounts, Currencies, Session) {
+function SimpleJournalVoucherController(AppCache, Vouchers, Accounts, Session, Notify) {
   var vm = this;
 
   // cache to save work-in-progress data and pre-fabricated templates
   var cache = AppCache('JournalVouchers');
 
   // bread crumb paths
-  vm.paths = [{
-    label : $translate.instant('VOUCHERS.SIMPLE.TITLE'),
-    current: true
-  }];
+  vm.paths = [
+    { label : 'TREE.FINANCE' },
+    { label : 'VOUCHERS.SIMPLE.TITLE' }
+  ];
 
-  // bind the startup method as a reset method
-  vm.reset = startup;
+  // bind the submit method
   vm.submit = submit;
 
   // load the list of accounts
-  Accounts.read(null, { detailed : 1 })
+  Accounts.read()
   .then(function (accounts) {
-    vm.accounts = removeChildren(accounts);
-  });
-
-  function removeChildren(accounts) {
-    var accountsOnly = accounts.map(function (item) {
-      delete item.children;
-      return item;
-    });
-    return accountsOnly;
-  }
-
-  // load the available currencies
-  Currencies.read()
-  .then(function (currencies) {
-
-    // format human readable labels
-    currencies.forEach(function (currency) {
-      currency.label = Currencies.format(currency.id);
-    });
-
-    // bind the currencies to the view
-    vm.currencies = currencies;
+    vm.accounts = accounts;
   });
 
-  /** run the module on startup and refresh */
+  /* run the module on startup and refresh */
   function startup() {
 
-    // delete any state indicators (if they exist)
-    delete vm.created;
-
     // current timestamp to limit date
     vm.timestamp = new Date();
 
     // set up default voucher values
     vm.voucher = {};
     vm.voucher.date = new Date();
-
+    vm.voucher.currency_id = Session.enterprise.currency_id;
   }
 
   function submit(form) {
-    // reset the state
-    vm.created = false;
-
-    // clear the old error if it exists
-    delete vm.httpError;
-
     // stop submission if the form is invalid
     if (form.$invalid) {
```

Comment:
It might make sense - I'll give it a try.

My original thinking was that the form is only five inputs long, so they should all be visible if validation occurs.  But if all other forms use `Notify.danger`, I'll conform to standards.